### PR TITLE
fixes #747, #748

### LINKED ISSUE #747: Weird things since upgrade to 3.4.4
[Sorry - don't get the issue template and can't assign any tags, for some reason. Hopefully someone can add for me please?]

I'm seeing two weird things since I upgraded to 3.4.4...

- NTP is disconnecting and reconnecting regularly, shown both in the system log and in HA
image
image
image
The gap in the bottom image on Saturday is when I did the upgrade, so the regular disconnects are only happening since then.

- Two new entities have appeared in HA...
image
image
This is curious, as I don't have a heat pump! There are no MQTT Explorer entries with the string 'heatpump' and no entities in HA with the string 'heatpump' either. So where the heck are these coming from? The logs include...
```
2022-11-14 07:51:37.758 WARNING (MainThread) [homeassistant.helpers.template] Template variable warning: 'dict object' has no attribute 'summermode' when rendering '{{value_json.hc1.summermode}}'
2022-11-14 08:00:19.147 WARNING (MainThread) [homeassistant.helpers.template] Template variable warning: 'dict object' has no attribute 'hpoperatingstate' when rendering '{{value_json.hc1.hpoperatingstate}}'
2022-11-14 08:05:48.332 WARNING (MainThread) [homeassistant.helpers.template] Template variable warning: 'dict object' has no attribute 'summermode' when rendering '{{value_json.hc1.summermode}}'
2022-11-14 08:16:11.731 WARNING (MainThread) [homeassistant.helpers.template] Template variable warning: 'dict object' has no attribute 'hpoperatingstate' when rendering '{{value_json.hc1.hpoperatingstate}}'
```

### PR DESCRIPTION
**Pull request by MichaelDvP**
- for #747 thetranslations are from google-translate, maybe needs rework from native speakers
- #748 is only the thermostat check for summer2_typeids, no feedback yet, but it should work
- i've forgot to add the "has_loader" in https://github.com/emsesp/EMS-ESP32/commit/1d4634a76cf0faa5a3cad0460b1a3311b0a45dd1
- changed the weblog to deque to free memory if buffersize is reduced. Now i check heap on every logmessage and size 20k. Have not tested if this is a good value. It's the size of the largest json buffer we use (EMSESP_JSON_SIZE_XXXLARGE_DYN)

### CODE REVIEW

**Comment by proddy** [comment]
> * changed the weblog to deque to free memory if buffersize is reduced. Now i check heap on every logmessage and size 20k. Have not tested if this is a good value. It's the size of the largest json buffer we use (EMSESP_JSON_SIZE_XXXLARGE_DYN)

Another thing on my list, not sure if mentioned it before, was to use https://github.com/muwerk/ustd to replace map, queues and lists eventually. It's light-weight and I used it in ESP v2 on the ESP8266, also contributed to the library back then. I think it should save more memory if heap becomes a problem. I just need to run some experiments as I'm afraid of breaking something.

File: src/web/WebLogService.cpp [hunk 1]
```diff
@@ -110,6 +110,10 @@ WebLogService::QueuedLogMessage::QueuedLogMessage(unsigned long id, std::shared_
 }
 
 void WebLogService::operator<<(std::shared_ptr<uuid::log::Message> message) {
+    if (ESP.getMaxAllocHeap() < 20480) {
```
Comment (proddy) [inline src/web/WebLogService.cpp:113]: i was was always worried these ESP.() functions are expensive and sometimes blocking, but if it works its good!

**Comment by MichaelDvP** [comment]
Free heap is already a problem for a few people with E32 and lot of entities from heatpump and RC300 with multiple hcs.
Typical issue is a white page for these pages that uses the 20k json. (also systeminfo only reports "NULL").

**Comment by proddy** [comment]
> Free heap is already a problem for a few people with E32 and lot of entities from heatpump and RC300 with multiple hcs. Typical issue is a white page for these pages that uses the 20k json. (also systeminfo only reports "NULL").

is it just the web UI that is impacted or does EMS-ESP crash and reboot? If it's just the web we can look at streaming the data on the Dashboard

**Comment by MichaelDvP** [comment]
I think only blank pages on web ui. One was in #748 after setting logbuffer to max. Another was on discord [here](https://discord.com/channels/816637840644505620/816958087160266775/1040371629764657173).

Testing the S2 with high logbuffer and growing mqtt queue also does not crash, but ems-esp was not reachable over wifi anymore (web and mqtt). With S32/E32 and c3 i always have enough memory in my system.

**Pull request merged**